.github/workflows/very-expensive-test.yml

```
@@ -0,0 +1,43 @@
+name: Very Expensive Test
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      # WARN: This will run very expensive test every time ANY label is added
+      - labeled
```

```suggestion
      # WARN: This will run very expensive test workflow every time ANY label is added (but we short circuit if there isn't the `need/very-expensive-test` label.
      - labeled
```

Isn't that right?  There's no real major downsides of running additionally and then short-circuiting right (at which point this is a good callout but not necessarily a major WARN to me).

---

The callout was more about the situation in which `need/very-expensive-test` label is present in the PR and then we add more labels. In that situation, whenever a new label is added, the tests are rerun. I don't think that's a blocker but I thought worth a callout.

---

Got it - makes sense.  We can resolve.